tests/test_pfcwd.py

```diff
@@ -0,0 +1,297 @@
+#                                      "BUFFER_POOL_WATERMARK_STAT_COUNTER:{}".format(buffer))
+#
+#        self.config_db.delete_entry("FLEX_COUNTER_TABLE", "PG_WATERMARK")
+#        self.config_db.delete_entry("FLEX_COUNTER_TABLE", "QUEUE_WATERMARK")
+#        self.config_db.delete_entry("FLEX_COUNTER_TABLE", "BUFFER_POOL_WATERMARK")
+#    
+
+    def test_ingress_pfcwd_handler(self, dvs, testlog):
+
+       try:
+            self.setup_dbs(dvs)
+            self.enable_device_pfcwd(dvs)
+            #self.enable_queue_counters(dvs)
+            self.set_admin_status("Ethernet64", "up")
+
+            pfcwd_info = {
+                    'detection_time': DEFAULT_DETECTION_TIME,
+                    'restoration_time': DEFAULT_RESTORATION_TIME,
+                    'action': DEFAULT_ACTION
+            }
+
+            self.config_db.update_entry(CONFIG_DB_PFC_WD_TABLE_NAME, "Ethernet64", pfcwd_info)
+            self.config_db.update_entry(CONFIG_DB_PFC_WD_TABLE_NAME, "GLOBAL", {'POLL_INTERVAL': '10'})
+            self.config_db.update_entry('FLEX_COUNTER_TABLE', "PFCWD", {'FLEX_COUNTER_STATUS': 'enable'})
+            
+            self.set_port_pfc(dvs, "Ethernet64", [3,4])
+
+            # Get SAI object ID for the interface
+            port_oid = self.get_port_oid(dvs, "Ethernet64")
+
+            # Verify default PFC is set to configured value
+            pfc = self.get_port_attr(dvs, port_oid, 'SAI_PORT_ATTR_PRIORITY_FLOW_CONTROL')
+            assert pfc == self.get_bitmask_str([3,4])
+
+            # Find OID for the queue 3
+            queue_oid = self.get_queue_oid(dvs, "Ethernet64:3")
+
+            self.set_up_flex_counter(dvs, port_oid, queue_oid)
+            self.enable_unittests(dvs, "true")
+            time.sleep(5)   
+
+#          self.verify_value(dvs, port_oid, "PFC_WD", "SAI_PORT_STAT_PFC_3_RX_PKTS", "0")
+#          self.verify_value(dvs, port_oid, "PFC_WD", "SAI_PORT_STAT_PFC_4_RX_PKTS", "0")
+#          self.verify_value(dvs, port_oid, "PFC_WD", "SAI_PORT_STAT_PFC_3_ON2OFF_RX_PKTS", "0")
+#          self.verify_value(dvs, port_oid, "PFC_WD", "SAI_PORT_STAT_PFC_4_ON2OFF_RX_PKTS", "0")
+#         
+#          self.verify_value(dvs, queue_oid, "PFC_WD", "SAI_QUEUE_STAT_PACKETS", "0")
+#          self.verify_value(dvs, queue_oid, "PFC_WD", "SAI_QUEUE_STAT_CURR_OCCUPANCY_BYTES", "0")
+#          self.verify_value(dvs, queue_oid, "PFC_WD", "SAI_QUEUE_ATTR_PAUSE_STATUS", "0")
+#          self.populate_asic_all(dvs, "SAI_OBJECT_TYPE_PORT", port_oid, "100")
+#          self.populate_asic_all(dvs, "SAI_OBJECT_TYPE_QUEUE", queue_oid, "100")
+            
+            self.start_pfcwd_on_all_ports(dvs)
+            # Enable PFC Storm on queue 3
+            self.set_pfc_storm_queue_oid(dvs, queue_oid)
+            
+            # Verify ASIC_DB for table creation
+            #verify_acl_table_creation(self, dvs)
+
+            #Verify ASIC_DB for rule creation
+            #verify_acl_rule_in_ports(dvs)
+
+       finally:
```

Comment:
please make sure to clean up all the objects that you created for this particular test